### PR TITLE
Unfatal Segfaults

### PR DESCRIPTION
**Pull request by elad335**
Segfaults on emulation threads will not terminate RPCS3 anymore.

### CODE REVIEW

File: Utilities/Thread.cpp [hunk 1]
```diff
@@ -1739,13 +1737,7 @@ static LONG exception_filter(PEXCEPTION_POINTERS pExp) noexcept
...
 	// TODO: print registers and the callstack
 
-	sys_log.fatal("\n%s", msg);
-
-	if (!IsDebuggerPresent())
-	{
-		report_fatal_error(msg);
-	}
-
+	thread_ctrl::emergency_exit(msg);
```
Comment (Nekotekina) [inline Utilities/Thread.cpp:1740]: Return after noreturn function.

File: Utilities/Thread.cpp [hunk 2]
```diff
@@ -1817,17 +1809,7 @@ static void signal_handler(int sig, siginfo_t* info, void* uct) noexcept
...
-
-	if (IsDebuggerPresent())
-	{
-		// Convert to SIGTRAP
-		raise(SIGTRAP);
-		return;
-	}
-
-	report_fatal_error(msg);
+	thread_ctrl::emergency_exit(msg);
```
Comment (Nekotekina) [inline Utilities/Thread.cpp:1812]: I'd recommend to be more careful with non-windows path. `raise(SIGTRAP)` was more tested than plain int3.

**Review by Nekotekina — CHANGES_REQUESTED** [review]

File: Utilities/Thread.cpp (hunk 2 above)
Comment (elad335) [reply Utilities/Thread.cpp:1821]: Okay, remained unchanged for now.

**Comment by Nekotekina** [comment]
Well, I can't disagree that calling abort/terminate in these situations might be overkill.

**Pull request merged**